EIPS/eip-1962.md

```diff
@@ -0,0 +1,190 @@
+---
+eip: 1962
+title: EC arithmetics and pairings with runtime definitions
+author: Alex Vlasov (@shamatar)
+discussions-to: https://github.com/ethereum/EIPs/pull/1962
+type: Standards Track
+category: Core
+status: Draft
+created: 2019-04-22
+requires: 1109
+---
+
+# Simple summary
+
+This proposal is an extension and formalization of EIP1829 with an inclusion of pairings. EIP1109 is required due to possibly very low cost of point addition operation compared to the STATICCALL opcode.
+
+## Abstract
+
+This EIP proposes the following actions to bring cryptographic functionality desired for privacy and scaling solutions:
+
+- Implement the following operations over elliptic curves in the Weierstrass form with curve parameters such as base field, A, B coefficients defined in runtime:
+    - Point addition
+    - Multiplication of a single point over a scalar
+    - Multiexponentiation
+- Implement pairing operation over elliptic curves from the following "families" with parameters such as base field, extension tower structure, coefficients defined in runtime:
+    - BLS12
+    - BN
+    - MNT4/6
+- Implement pairing operation over elliptic curves found by Cocks-Pinch method. Limit k <= 8.  
+
+## Motivation
+
+- Proposal to implement base elliptic curve arithmetics is covered by EIP1829 and will allow to implement various privacy-presenving protocols with a reasonable gas costs per operation.
+- Pairings:
+    - Extend set of curve that are available to allow Ethereum users to choose their security parameters and required functionality.
+    - Due to large variety of curves implementation of this EIP should cover as broad area as possible to allow Ethereum users to experiment with their choices and constructions without waiting for new forks.
+- Gas costs - this EIP is designed to also specify parameters and approach to gas cost evaluation before-hand. This is a strict requirement for any precompile to allow Ethereum nodes to efficiently reject transactions and operations as early as possible.
+
+## Specification
+This specification will not yet provide concrete numbers for every operation, but will first intoduce interfaces and gas calculation approaches.
```

Comment:
Is this introducing a new precompile? Is it extending an existing one? Please state, even with a placeholder pecompile address, if this is a new precompile.